Can't include eigen.h and operators.h at the same time

If I include both `pybind11/eigen.h` and `pybind11/operators.h` at the same time, I get a compilation failure:

```
[  3%] Building CXX object tests/CMakeFiles/pybind11_tests.dir/test_eigen.cpp.o
In file included from /home/jagerman/research/pyeris/pybind11/include/pybind11/eigen.h:12:0,
                 from /home/jagerman/research/pyeris/pybind11/tests/test_eigen.cpp:12:
/home/jagerman/research/pyeris/pybind11/include/pybind11/numpy.h: In static member function ‘static void pybind11::detail::npy_format_descriptor<T, typename std::enable_if<pybind11::detail::is_pod_struct<type>::value>::type>::register_dtype(std::initializer_list<pybind11::detail::field_descriptor>)’:
/home/jagerman/research/pyeris/pybind11/include/pybind11/numpy.h:410:45: error: invalid initialization of reference of type ‘const pybind11::detail::self_t&’ from expression of type ‘size_t {aka long unsigned int}’
             offsets.append(int_(field.offset));
                                             ^
In file included from /home/jagerman/research/pyeris/pybind11/tests/test_eigen.cpp:11:0:
/home/jagerman/research/pyeris/pybind11/include/pybind11/operators.h:143:37: note: in passing argument 1 of ‘pybind11::detail::op_<(pybind11::detail::op_id)16, (pybind11::detail::op_type)2, pybind11::detail::self_t, pybind11::detail::undefined_t> pybind11::detail::int_(const pybind11::detail::self_t&)’
 PYBIND11_UNARY_OPERATOR(int,        int_,         (int) l)
                                     ^
/home/jagerman/research/pyeris/pybind11/include/pybind11/operators.h:103:48: note: in definition of macro ‘PYBIND11_UNARY_OPERATOR’
 inline op_<op_##id, op_u, self_t, undefined_t> op(const self_t &) {                    \
                                                ^~
/home/jagerman/research/pyeris/pybind11/tests/test_eigen.cpp: In function ‘void init_eigen(pybind11::module&)’:
/home/jagerman/research/pyeris/pybind11/tests/test_eigen.cpp:15:29: warning: unused parameter ‘m’ [-Wunused-parameter]
 void init_eigen(py::module &m) {
                             ^
tests/CMakeFiles/pybind11_tests.dir/build.make:638: recipe for target 'tests/CMakeFiles/pybind11_tests.dir/test_eigen.cpp.o' failed
make[2]: *** [tests/CMakeFiles/pybind11_tests.dir/test_eigen.cpp.o] Error 1
CMakeFiles/Makefile2:86: recipe for target 'tests/CMakeFiles/pybind11_tests.dir/all' failed
make[1]: *** [tests/CMakeFiles/pybind11_tests.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
```

The error above occurs when adding `#include <pybind/operators.h>` into `test_eigen.cpp`, but occurs even if adding the header but removing all the actual code from `test_eigen.cpp` (i.e. with everything deleted except for an empty-body `void init_eigen(py::module &) {}` function).

--- a/include/pybind11/numpy.h
+++ b/include/pybind11/numpy.h
@@ -407,7 +407,7 @@ struct npy_format_descriptor<T, typename std::enable_if<is_pod_struct<T>::value>
                 pybind11_fail("NumPy: unsupported field dtype");
             names.append(PYBIND11_STR_TYPE(field.name));
             formats.append(field.descr);
-            offsets.append(int_(field.offset));
+            offsets.append(pybind11::int_(field.offset));
         }
         dtype_ptr = pybind11::dtype(names, formats, offsets, sizeof(T)).release().ptr();
 

